app/src/main/java/com/teamagam/gimelgimel/app/common/logging/Log4jDiskLogger.java

```diff
@@ -0,0 +1,221 @@
+package com.teamagam.gimelgimel.app.common.logging;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+
+import java.io.File;
+
+import de.mindpipe.android.logging.log4j.LogConfigurator;
+
+
+class Log4jDiskLogger extends BaseLifecycleLogger {
+
+    private static final String VERBOSITY_USER_INTERACTION = "USER_INTERACTION";
+    private static final String VERBOSITY_LIFECYCLE = "LIFECYCLE";
+
+    private Logger mLogger;
+
+    Log4jDiskLogger(String directory, String fileName, long maxFileSize, int maxBackupLogFiles, String tag) {
+        try {
+            mLogger = createLogger(directory, fileName, maxFileSize, maxBackupLogFiles, tag);
+        } catch (Exception ex) {
+            mLogger = null;
+        }
+    }
+
+    private Logger createLogger(String directory, String fileName, long maxFileSize, int maxBackupLogFiles, String tag) {
+        createDirectory(directory);
+        configureLogger(directory, fileName, maxFileSize, maxBackupLogFiles);
```

Comment:
It seems this method configures not just this logger, but any log4j logger. Hence, this method should also be called once. !fix
[and perhaps get a different name, to indicate that fact]